src/style-spec/function/convert.js

```diff
@@ -153,7 +153,18 @@ function convertPropertyFunction(parameters, propertySpec, stops) {
         ];
     } else if (type === 'exponential') {
         const base = parameters.base !== undefined ? parameters.base : 1;
-        const expression = [getInterpolateOperator(parameters), ['exponential', base], ['number', get]];
+        const expression =
+            base === 1 ?
+                [
+                    getInterpolateOperator(parameters),
+                    ["linear"],
+                    ["number", get]
+                ] :
+                [
+                    getInterpolateOperator(parameters),
+                    ["exponential", base],
+                    ["number", get]
+                ];
```

Comment:
Small nit: we could make this a more compact by using the ternary only for the `["linear"]` and `["exponential", base]` so that there's less duplication.